modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -553,10 +553,10 @@ class ConvolutionLayerImpl CV_FINAL : public BaseConvolutionLayerImpl
             }
             else
             {
-                Mat newWeights = blobs[0].reshape(1, outCn);
-                Mat cvWeights = weightsMat.colRange(0, newWeights.cols);
+                Mat newWeights;
+                Mat cvWeights = weightsMat.colRange(0, blobs[0].total() / outCn);
                 cvWeights.copyTo(newWeights);
-                ieWeights = std::make_shared<ngraph::op::Constant>(ngraph::element::f32, kernel_shape, blobs[0].data);
+                ieWeights = std::make_shared<ngraph::op::Constant>(ngraph::element::f32, kernel_shape, newWeights.data);
```

Comment:
Can we use `cvWeights` instead `newWeights`?

Reply:
cvWeight is discontinuous so we need to make a copy of it's ROI